generator/docs/CombinationStrategies.md

```diff
@@ -7,7 +7,7 @@ In multiple areas of the generator, we find ourselves with multiple streams of v
 
 There are multiple ways to perform this selection process; we refer to these as **combination strategies**. Different combination strategies have different properties.
 
-Our current default is the **pinning** strategy.
+Our current default is the **exhaustive** strategy.
```

Comment:
combination strategies are weird atm

they are used in two places.

once exclusively in the cartesian walker, where it combines values for fields together.

once when partitioning, the partitions need combining together. (this is for reductive and cartesian)

when running in random generation mode (the current default) we use the minimal combination strategy. this is to enable all the values to keep changing, rather than trying to use the same value over and over

when not in random mode, then a person can choose between the different combination strategies. the default then being exhaustive.